satip/eumetsat.py

```diff
@@ -679,3 +687,63 @@ def eumetsat_cloud_name_to_datetime(filename: str):
     """Takes a file from the EUMETSAT API and returns the it's datetime part for Cloud mask files"""
     date_str = filename.split("0100-0100-")[-1].split(".")[0]
     return datetime.datetime.strptime(date_str, "%Y%m%d%H%M%S")
+
+
+class DownloadManager:
+    """
+    Main download manager class to handle both EUMETSAT
+
+    and GOES data downloading based on the provider.
+    """
+
+    def __init__(self, provider, user_key=None,
+                user_secret=None, data_dir=None,
+                log_directory=None):
+        """
+        Initialize the DownloadManager.
+
+        Args:
+            provider (str): Provider name ('EUMETSAT' or 'GOES').
+            user_key (str): User key for accessing data (for EUMETSAT).
+            user_secret (str): User secret for accessing data (for EUMETSAT).
+            data_dir (str): Directory to save downloaded data.
+            log_directory (str): Directory to save logs.
+        """
+        self.provider = provider
+
+        if self.provider == "EUMETSAT":
+            self.download_manager = EUMETSATDownloadManager(user_key, user_secret,
+                                                            data_dir, log_directory)
+        elif self.provider == "GOES":
+            self.download_manager = GOESDownloadManager(data_dir, log_directory)
+        else:
+            raise ValueError("Invalid provider. Supported providers are 'EUMETSAT' and 'GOES'.")
+
+    def download_data(self, start_time, end_time):
+        """
+        Download data for the specified time range.
+
+        Args:
+            start_time (datetime): Start of the download period.
+            end_time (datetime): End of the download period.
+        """
+        if self.provider == "GOES":
+            self.download_manager.download_goes_data(start_time, end_time)
+
+
+# Example usage
+if __name__ == "__main__":
```

Comment:
This is good! For the example, I would put it in the docstring, or in the README instead. The code here in these files should just be the classes and functions. Just changing it like this will break other parts of the code, so either other ones should be updated to call the EUMETSATDownloadManager instead.